src/libraries/System.Security.Cryptography.Cose/tests/SignVerify.AlgorithmSupport.cs

```diff
@@ -0,0 +1,241 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#pragma warning disable SYSLIB5006
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using Test.Cryptography;
+using Xunit;
+
+using System.Formats.Cbor;
+using System.Linq;
+
+namespace System.Security.Cryptography.Cose.Tests
+{
+    public class SignVerify
+    {
+        [ConditionalFact(typeof(MLDsa), nameof(MLDsa.IsSupported))]
+        public static void DecodeSign1DraftExampleMLDsa44()
+        {
+            using MLDsa pubKey = MLDsa.ImportMLDsaPublicKey(MLDsaAlgorithm.MLDsa44, MLDsa44PkHex.HexToByteArray());
+
+            CoseSign1Message msg = CoseMessage.DecodeSign1(MLDsa44Sign1Hex.HexToByteArray());
+            Assert.NotNull(msg.Content);
+            Assert.True(msg.VerifyEmbedded(CreateMLDsaAsymmetricAlgorithm(pubKey)));
+
+            Assert.Equal(2, msg.ProtectedHeaders.Count);
+            Assert.Equal(0, msg.UnprotectedHeaders.Count);
+
+            if (msg.ProtectedHeaders.TryGetValue(CoseHeaderLabel.Algorithm, out var alg))
+            {
+                Assert.Equal(-48 /* MLDsa44 */, alg.GetValueAsInt32());
+            }
+
+            if (msg.ProtectedHeaders.TryGetValue(CoseHeaderLabel.KeyIdentifier, out var keyIdentifier))
+            {
+                Assert.Equal(32 /* can't correlate that with anything on the draft example, it doesn't seem to match kid */, keyIdentifier.GetValueAsBytes().Length);
```

Comment:
I'm still confused where is this value coming from - it supposedly should be somewhere in the draft